embedding/csharp/Tizen.Flutter.Embedding/NUIFlutterView.cs

```diff
@@ -0,0 +1,199 @@
+    /// </summary>
+    public class NUIFlutterView : ImageView
+    {
+        /// <summary>
+        /// The Flutter engine instance.
+        /// </summary>
+        public FlutterEngine Engine { get; set; } = null;
+
+        /// <summary>
+        /// The Flutter view instance handle.
+        /// </summary>
+        private FlutterDesktopView View { get; set; } = new FlutterDesktopView();
+
+        /// <summary>
+        /// Whether the view is running.
+        /// </summary>
+        public bool IsRunning => !View.IsInvalid;
+
+        /// <summary>
+        /// When the view last received a touch event in milliseconds.
+        /// </summary>
+        private uint _lastTouchEventTime = 0;
+
+        /// <summary>
+        /// The size of the Flutter view.
+        /// </summary>
+        private Size2D _size = new Size2D();
+
+        /// <summary>
+        /// Starts running the view with the associated engine, creating if not set.
+        /// </summary>
+        /// <remarks>
+        /// <see cref="Engine"/> must not be set again after this call.
+        /// <see cref="Destroy"/> must be called if the view is no longer used.
+        /// </remarks>
+        public bool RunEngine()
+        {
+            if (IsRunning)
+            {
+                TizenLog.Error("The engine is already running.");
+                return false;
+            }
+
+            if (Engine == null)
+            {
+                Engine = new FlutterEngine();
+            }
+
+            if (!Engine.IsValid)
+            {
+                TizenLog.Error("Could not create a Flutter engine.");
+                return false;
+            }
+
+            _size = GetDefaultSize();
+
+            Type baseType = typeof(NativeImageQueue).BaseType.BaseType.BaseType;
+            FieldInfo field = baseType.GetField("swigCPtr", BindingFlags.NonPublic | BindingFlags.Instance);
+            NativeImageQueue nativeImageQueue =
+                new NativeImageQueue((uint)_size.Width, (uint)_size.Height, NativeImageQueue.ColorFormat.RGBA8888);
+            HandleRef nativeImageQueueRef = (HandleRef)field.GetValue(nativeImageQueue);
+            SetImage(nativeImageQueue.GenerateUrl().ToString());
+
+            Type imageViewBaseType = typeof(ImageView).BaseType.BaseType.BaseType.BaseType;
+            FieldInfo imageViewField =
+                imageViewBaseType.GetField("swigCPtr", BindingFlags.NonPublic | BindingFlags.Instance);
+            HandleRef imageViewRef = (HandleRef)imageViewField.GetValue(this);
+
+            var viewProperties = new FlutterDesktopViewProperties
+            {
+                width = _size.Width,
+                height = _size.Height,
+            };
+
+            View = FlutterDesktopViewCreateFromImageView(
+                ref viewProperties, Engine.Engine, imageViewRef.Handle, nativeImageQueueRef.Handle,
+                NUIApplication.GetDefaultWindow().GetNativeId());
```

Comment:
`GetDefaultWindow` seems to be an internal API (it has an `EditorBrowsableState.Never` attribute).

```suggestion
                Window.Instance.GetNativeId());
```

Please also check the line 128.

Reply:
wow.. thanks..